src/pages/home/library/Discography.tsx

```diff
@@ -2,23 +2,54 @@ import { Box } from "@mui/material";
 import { FunctionComponent, useEffect, useMemo, useState } from "react";
 import theme from "theme";
 import { SearchBox } from "components";
-import { Song, useGetSongsQuery, useHlsJs } from "modules/song";
+import {
+  Song,
+  useGetSongCountQuery,
+  useGetSongsQuery,
+  useHlsJs,
+} from "modules/song";
 import { TableSkeleton, Typography } from "elements";
 import { useWindowDimensions } from "common";
 import NoSongsYet from "./NoSongsYet";
 import SongList from "./SongList";
 
 const Discography: FunctionComponent = () => {
-  const {
-    data: songs = [],
-    isLoading,
-    isSuccess,
-  } = useGetSongsQuery({ ownerIds: ["me"] });
+  const headerHeight = 245;
+  const footerHeight = 40;
+  const bottomPadding = 30;
+  const rowHeight = 65;
+  const viewportWidth = useWindowDimensions()?.width;
+  const viewportHeight = useWindowDimensions()?.height;
+
   const [filteredData, setFilteredData] = useState<Song[]>([]);
   const [query, setQuery] = useState("");
   const [page, setPage] = useState(1);
+  const [rowsPerPage, setRowsPerPage] = useState(1);
   const [currentPlayingSongId, setCurrentPlayingSongId] = useState<string>();
-  const viewportWidth = useWindowDimensions()?.width;
+
+  let songsToRequest = rowsPerPage;
+  const { data: { count: totalCountOfSongs = 0 } = {} } = useGetSongCountQuery({
+    ownerIds: ["me"],
+  });
+
+  const totalPagesCount = Math.ceil(totalCountOfSongs / rowsPerPage);
+  const remainingSongsOnLastPage = totalCountOfSongs % rowsPerPage;
+
+  // Determines how many songs to request for the last page
+  if (page === totalPagesCount) {
+    songsToRequest =
+      remainingSongsOnLastPage > 0 ? remainingSongsOnLastPage : rowsPerPage;
+  }
+
+  const {
+    data: songs = [],
+    isLoading,
+    isSuccess,
+  } = useGetSongsQuery({
+    ownerIds: ["me"],
+    offset: page - 1,
+    limit: songsToRequest,
+  });
```

Comment:
Considering putting the fetching logic in the SongList and passing a `query` prop to add a `phrase` here so it makes it easier for rerendering on search of songs. Will revisit tomorrow

Reply:
Should be fine here for now, refactor on search implementation. Maybe I don't see the current optimization of this right now.